plugins/modules/mso_schema_template_anp_epg_useg_attribute.py

```diff
@@ -0,0 +1,283 @@
+        operator=dict(type="str", choices=["equals", "contains", "starts_with", "ends_with"]),
+        useg_subnet=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "query", ["schema", "template", "anp", "epg"]],
+            ["state", "absent", ["schema", "template", "anp", "epg", "name"]],
+            ["state", "present", ["schema", "template", "anp", "epg", "name", "attribute_type"]],
+            ["useg_subnet", False, ["value"]],
+        ],
+    )
+
+    schema = module.params.get("schema")
+    template = module.params.get("template").replace(" ", "")
+    anp = module.params.get("anp")
+    epg = module.params.get("epg")
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+
+    attribute_type = module.params.get("attribute_type")
+    value = module.params.get("value")
+    operator = module.params.get("operator")
+
+    useg_subnet = module.params.get("useg_subnet")
+
+    state = module.params.get("state")
+
+    mso = MSOModule(module)
+    if state == "present":
+        if attribute_type in ["mac", "dns"] and value is None:
+            mso.fail_json(msg="Failed due to invalid 'value' and the attribute_type is: {0}.".format(attribute_type))
+        elif attribute_type in [
+            "vm",
+            "vm_datacenter",
+            "vm_hypervisor_identifier",
+            "vm_operating_system",
+            "vm_tag",
+            "vm_identifier",
+            "vmm_domain",
+            "vnic_dn",
+            "vm_name",
+        ] and (value is None or operator is None):
+            mso.fail_json(msg="Failed due to invalid 'value' or 'operator' and the attribute_type is: {0}.".format(attribute_type))
+
+        if attribute_type == "ip" and useg_subnet is False:
+            fvSubnet = True
+        elif attribute_type == "ip" and useg_subnet is True:
+            value = "0.0.0.0"
+            fvSubnet = False
+
+    # Get schema
+    schema_id, schema_path, schema_obj = mso.query_schema(schema)
+
+    # Get template
+    templates = [t.get("name") for t in schema_obj.get("templates")]
+    if template not in templates:
+        mso.fail_json(msg="Provided Template: {0} does not exist.".format(template))
+    template_idx = templates.index(template)
+
+    # Get ANP
+    anps = [a.get("name") for a in schema_obj.get("templates")[template_idx]["anps"]]
+    if anp not in anps:
+        mso.fail_json(msg="Provided Application Profile: {0} does not exist.".format(anp))
+    anp_idx = anps.index(anp)
+
+    # Get EPG
+    epgs = [e.get("name") for e in schema_obj.get("templates")[template_idx]["anps"][anp_idx]["epgs"]]
+    if epg not in epgs:
+        mso.fail_json(msg="Provided EPG: {0} does not exist.".format(epg))
+    epg_idx = epgs.index(epg)
```

Comment:
This can be rewritten using the MSOSchema class. This would impact the rest of your logic.

```suggestion
mso_schema = MSOSchema(mso, schema, template)
mso_schema.set_template_anp(anp)
mso_schema.set_template_anp_epg(epg)
```